src/k2/transport/RPCDispatcher.h

```diff
@@ -243,12 +247,15 @@ class RPCDispatcher: public seastar::weakly_referencable<RPCDispatcher> {
     void _handleNewMessage(Request&& request);
 
     // Helper method useds to send messages
-    void _send(Verb verb, std::unique_ptr<Payload> payload, TXEndpoint& endpoint, MessageMetadata meta);
+    seastar::future<> _send(Verb verb, std::unique_ptr<Payload> payload, TXEndpoint& endpoint, MessageMetadata meta);
 
 private: // fields
     // the protocols this dispatcher will be able to support
     std::unordered_map<String, seastar::shared_ptr<IRPCProtocol>> _protocols;
 
+    // add the map for core-address
```

Comment:
use the comment to say what we need this for, not what it is. So instead of saying its a map of endpoint-to-core, say that this is the mapping we use to discover what URL is owned by what core in order to perform xcore dispatch

Reply:
Wrote an understandable comment.